frontend/src/pages/PipelineDetails.tsx

```diff
@@ -248,34 +290,72 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
+      } catch (err) {
+        await this.showPageError('Cannot retrieve run details.', err);
+        logger.error('Cannot retrieve run details.', err);
+        return;
+      }
+    } else if (fromRecurringRunId) {
+      try {
+        const recurringRunDetails = await Apis.jobServiceApi.getJob(fromRecurringRunId);
+        origin = {
+          isRecurring: true,
+          recurringRun: recurringRunDetails,
+        };
 
-        // Build the breadcrumbs, by adding experiment and run names
-        if (experiment) {
-          breadcrumbs.push(
-            { displayName: 'Experiments', href: RoutePage.EXPERIMENTS },
-            {
-              displayName: experiment.name!,
-              href: RoutePage.EXPERIMENT_DETAILS.replace(
-                ':' + RouteParams.experimentId,
-                experiment.id!,
-              ),
-            },
-          );
+        if (
+          isFeatureEnabled(FeatureKey.V2_ALPHA) &&
+          recurringRunDetails.pipeline_spec?.pipeline_manifest
+        ) {
+          templateString = recurringRunDetails.pipeline_spec.pipeline_manifest;
         } else {
-          breadcrumbs.push({ displayName: 'All runs', href: RoutePage.RUNS });
+          try {
+            const workflowManifestString = RunUtils.getWorkflowManifest(recurringRunDetails) || '';
+            const workflowManifest = JSON.parse(workflowManifestString || '{}');
+            try {
+              if (WorkflowUtils.isPipelineSpec(workflowManifestString)) {
+                templateString = workflowManifestString;
+              } else {
+                templateString = JsYaml.safeDump(workflowManifest);
+              }
+            } catch (err) {
+              await this.showPageError(
+                `Failed to parse pipeline spec from run with ID: ${recurringRunDetails.id}.`,
+                err,
+              );
+              logger.error(
+                `Failed to convert pipeline spec YAML from run with ID: ${recurringRunDetails.id}.`,
+                err,
+              );
+            }
+          } catch (err) {
+            await this.showPageError(
+              `Failed to parse pipeline spec from run with ID: ${recurringRunDetails.id}.`,
+              err,
+            );
+            logger.error(
+              `Failed to parse pipeline spec JSON from run with ID: ${recurringRunDetails.id}.`,
+              err,
+            );
+          }
         }
-        breadcrumbs.push({
-          displayName: runDetails.run!.name!,
-          href: RoutePage.RUN_DETAILS.replace(':' + RouteParams.runId, fromRunId),
-        });
+
+        const relatedExperimentId = RunUtils.getFirstExperimentReferenceId(recurringRunDetails);
+        let experiment: ApiExperiment | undefined;
+        if (relatedExperimentId) {
+          experiment = await Apis.experimentServiceApi.getExperiment(relatedExperimentId);
+        }
+        this.buildBreadCumb(breadcrumbs, origin, fromRecurringRunId, experiment);
```

Comment:
Why are we passing `fromRecurringRunId` to `runId` argument? RecurringRunId and RunId are ID for different database tables.